### PR TITLE
CPO Endpoints FTAs

### PR DESCRIPTION
**Pull request by noronhaa**
FTAs for the CREATE, GET, UPDATE, DELETE endpoints

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CPO-30

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    When a request is prepared with appropriate values
+    And the request [intends to Create a case payment order]
+    And the request [does not contain one or more of the mandatory parameters]
+    And the request [does not contain the <parameter> parameter]
+    And it is submitted to call the [createCasePaymentOrder] operation of [Case Payment Orders Microservice],
+    Then a negative response is received
+    And the response has all other details as expected
+    And a call [to verify that a Case payment Order has not been created in the database] will get the expected response as in [getCasePaymentOrder_cpo_not_created].
+
+  @Ignore #@S-701.2.1 TODO re-enable after CPO-40 is fixed
```
Comment (sullivangareth) [inline src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature:34]: CPO-40 is marked as "Withdrawn" in Jira ?

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  Examples:
+    | parameter        |
+    | effective_from   |
```
Comment (sullivangareth) [inline src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature:37]: `effective_from` will be removed in CPO-36

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/S-701.2.2.td.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  "specs": [
+    "does not contain the case_id parameter"
+  ],
+
+  "request" : {
+    "_extends_": "Common_Request",
+    "body": {
+      "effective_from" : "2021-01-01T00:00:00Z",
+      "case_type_id" : "FT_MasterCaseType",
```
Comment (sullivangareth) [inline src/functionalTest/resources/features/F-701 - createCasePaymentOrder/S-701.2.2.td.json:13]: Not required ?

File: src/functionalTest/resources/features/F-702 - updateCasePaymentOrder/F-702.feature [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    And a successful call [to create a case payment order for the created case] as in [Prerequisite_Create_CPO],
+
+  @S-702.1
+  Scenario: AC1: Successfully allow the update of a previously created payment order in the Case Payment Order database
+    When a request is prepared with appropriate values
+    And the request [intends to update a previously created payment order]
+    And the request [contains all the mandatory parameters]
+    And it is submitted to call the [updateCasePaymentOrder] operation of [Case Payment Orders Microservice]
+    Then a positive response is received
+    And the response [contains a 200 success OK code]
```
Comment (sullivangareth) [inline src/functionalTest/resources/features/F-702 - updateCasePaymentOrder/F-702.feature:16]: I think the Update response is now a 202 ?

File: src/functionalTest/resources/features/F-703 - getCasePaymentOrder/F-703.feature [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    Then a positive response is received
+    And the response [contains a 200 success OK code]
+    And the response [case payment order CP1, CP2]
+    And the response [has page_size default 20 and page_number default 1]
+    And the response has all other details as expected
+
+  @S-703.3
+  Scenario: AC3: Mandatory parameters missing from the request (Search Criteria Missing)
+    And a successful call [to create a case payment order CP1] as in [Prerequisite_Create_CPO],
+    And a successful call [to create a case payment order CP2] as in [Prerequisite_Create_CPO_2],
```
Comment (sullivangareth) [inline src/functionalTest/resources/features/F-703 - getCasePaymentOrder/F-703.feature:41]: Do we need to create these case payment orders ?

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  Scenario: AC4a- Must return error if order_reference for a case_id is non-unique
+    And a successful call [to create a case payment order for the created case] as in [Prerequisite_Create_CPO],
+    When a request is prepared with appropriate values
+    And the request [intends to Create a case payment order]
+    And the request [contains all the mandatory parameters]
+    And the request [contains the same order_reference as used in the previously created CPO for the same case id]
+    And it is submitted to call the [createCasePaymentOrder] operation of [Case Payment Orders Microservice],
+    Then a negative response is received
+    And the response has all other details as expected
+#    And a call [to verify that a Case payment Order has not been created in the database which can be accessed or queried again at a later date] will get the expected response as in [CPO-5_getCasePaymentOrder_cpo_not_created].
```
Comment (sullivangareth) [inline src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature:95]: can commented out line be deleted ?

**Review by sullivangareth — COMMENTED** [review]
Main comment is effective_from will soon be removed when CPO-36 is merged, but I guess there'll be another PR to address that ?

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature (hunk 1 above)
Comment (noronhaa) [reply src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature:34]: correct, can remove this scenario now

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/S-701.2.2.td.json (hunk 3 above)
Comment (noronhaa) [reply src/functionalTest/resources/features/F-701 - createCasePaymentOrder/S-701.2.2.td.json:13]: removed from test and other test files

File: src/functionalTest/resources/features/F-703 - getCasePaymentOrder/F-703.feature (hunk 5 above)
Comment (noronhaa) [reply src/functionalTest/resources/features/F-703 - getCasePaymentOrder/F-703.feature:41]: cpo's removed

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature (hunk 6 above)
Comment (noronhaa) [reply src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature:95]: yes- removed

File: src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature (hunk 2 above)
Comment (noronhaa) [reply src/functionalTest/resources/features/F-701 - createCasePaymentOrder/F-701.feature:37]: test removed accordingly

File: src/functionalTest/resources/features/F-702 - updateCasePaymentOrder/F-702.feature (hunk 4 above)
Comment (noronhaa) [reply src/functionalTest/resources/features/F-702 - updateCasePaymentOrder/F-702.feature:16]: amended

**Review by sullivangareth — APPROVED** [review]

**Pull request merged**